### DISCUSSION TITLE
Multiple authentication mechanisms

### DISCUSSION THREAD

**Discussion by pnat2024** [apiflask/apiflask#535] [Q&A]
Newbie here -- Is there a way to enable multiple auth mechanisms in APIFlask -- for example get X-API-Key or X-API-Token from the header whichever is available? I think this is similar to the MultiAuth mechanism in Flask-HttpAuth https://github.com/miguelgrinberg/Flask-HTTPAuth/blob/main/examples/multi_auth.py

Thanks!

**Comment by NicolaCortese** [comment]
Looks like using the https://flask-httpauth.readthedocs.io/en/latest/#flask_httpauth.HTTPMultiAuth might be a solution, but in trying to  implement it I broke my swagger page as it doesn't seem to support it..
